### PR TITLE
stabilize cfg_attr_multi

### PR DESCRIPTION
**Pull request by Dylan-DPC-zz**
Stabilizes cfg_attr_multi feature

Related to #54881 

Will add the lint in a seperate PR

r? @Centril

### CODE REVIEW

File: src/test/ui/conditional-compilation/cfg-attr-multi-false.rs [hunk 1]
```diff
@@ -4,7 +4,7 @@
 // compile-pass
 
 #![warn(unused_must_use)]
-#![feature(cfg_attr_multi)]
+#![cfg_attr(stage0, feature(cfg_attr_multi))]
```
Comment (petrochenkov) [inline src/test/ui/conditional-compilation/cfg-attr-multi-false.rs]: `cfg_attr(stage0, ...)` is not required in tests, the features can just be removed.

File: src/test/ui/conditional-compilation/cfg-attr-parse.rs [hunk 2]
```diff
@@ -1,6 +1,6 @@
 // Parse `cfg_attr` with varying numbers of attributes and trailing commas
 
-#![feature(cfg_attr_multi)]
+#![cfg_attr(stage0, feature(cfg_attr_multi))]
```
Comment (Centril) [inline src/test/ui/conditional-compilation/cfg-attr-parse.rs]: Don't think this is needed as well.

File: src/libsyntax/feature_gate.rs [hunk 3]
```diff
@@ -693,6 +690,8 @@ declare_features! (
     (accepted, repr_packed, "1.33.0", Some(33158), None),
     // Allows calling `const unsafe fn` inside `unsafe` blocks in `const fn` functions.
     (accepted, min_const_unsafe_fn, "1.33.0", Some(55607), None),
+// `#[cfg_attr(predicate, multiple, attributes, here)]`
```
Comment (Centril) [inline src/libsyntax/feature_gate.rs]: Style nit: indentation doesn't fit with the rest of the lines.

**Comment by Centril** [comment]
You have some merge conflicts that'll need to be resolved; other than that it looks good.

**Comment by Dylan-DPC** [comment]
Damn some issue with the rebase.. let me try again

File: src/test/ui/conditional-compilation/cfg-attr-multi-invalid-2.rs [hunk 4]
```diff
@@ -1,7 +1,6 @@
 //
 // compile-flags: --cfg broken
 
-#![feature(cfg_attr_multi)]
```
Comment (Centril) [inline src/test/ui/conditional-compilation/cfg-attr-multi-invalid-2.rs]: Since this test is meant to fail, maybe you need to `--bless` it? (and possibly the `-invalid-1.rs` file as well)

**Review by Centril — APPROVED** [review]

**Comment by Centril** [comment]
Looks good; thanks!

Giving prio to this since it's a stabilization.

@bors r+ p=1

**Comment by Xanewok** [comment]
I believe this is spurious, (un?)fortunately:

```
thread 'cmd_changing_workspace_lib_retains_bin_diagnostics' panicked at 'expected bin error diagnostic', src/libcore/option.rs:1040:5
```
